src/rest/impl/deliveries.service.ts

```diff
@@ -132,16 +133,21 @@ export default class DeliveriesServiceImpl extends DeliveriesService {
  
       if (!deliveryContactSapId) {
         this.sendBadRequest(res, `Missing sapId on delivering user ${deliveryContact.id}`);
-        return;  
+        return;
       }
 
       if (!sapSalesPersonCode) {
         this.sendBadRequest(res, `Missing sapId on receiving user ${receivingContact.id}`);
-        return;  
+        return;
       }
 
       const databaseDelivery = await models.createDelivery(uuid(), productId, userId, time, status, amount, price, unitPrice, unitPriceWithBonus, qualityId, databaseDeliveryPlace.id);
-      await this.buildPurchaseXML(databaseDelivery, product, databaseDeliveryPlace, unitPriceWithBonus, deliveryContactSapId, sapSalesPersonCode, req.body.loans || [], itemGroup.category);
+
+      try {
+        await this.createDeliveryDocumentsToSap(databaseDelivery, product, databaseDeliveryPlace, unitPriceWithBonus, deliveryContactSapId, sapSalesPersonCode, req.body.loans || [], itemGroup.category);
+      } catch (e) {
+        getLogger().error(e);
```

Comment:
Shoudn't this respond with non-success status code, when operation fails?

Reply:
I thought that because the delivery is already in our database, it should not respond with direct error anymore, but log the error that it has failed to create documents to SAP side. It can return non-success code, but it might be a bit confusing in client side, when the contract still goes to our database even when this causes the response to be 500.